codex-ui/src/vue/components/alert/core/useAlert.ts

```diff
@@ -0,0 +1,96 @@
+import { ref } from 'vue';
+import { createSharedComposable } from '@vueuse/core';
+import type { AlertInterface, AlertOptions, AlertType } from './types';
+import { getId } from './constant';
+
+export const useStore = createSharedComposable(() => {
+  const alertRef = ref<HTMLElement | null>(null);
+  const alertStore = ref<AlertOptions[]>([]);
+
+  /**
+   * trigger alert component
+   * @param opt - alert options
+   */
+  function createAlertHandler(opt: Pick<AlertOptions, 'id' | 'icon' | 'message' | 'timeout' | 'type'>): void {
+    opt.id = getId();
+    const alert = alertStore.value.findIndex((idx: AlertOptions) => idx.id === opt.id);
+
+    alertStore.value.push(opt);
+
+    setTimeout(() => {
+      alertStore.value.splice(alert, alertStore.value.length);
+    }, opt.timeout);
+  }
+
+  function success(opt: AlertOptions): void {
+    createAlertHandler(opt);
+  }
+
+  function error(opt: AlertOptions): void {
+    createAlertHandler(opt);
+  }
+
+  function warning(opt: AlertOptions): void {
+    createAlertHandler(opt);
+  }
+
+  function info(opt: AlertOptions): void {
+    createAlertHandler(opt);
+  }
+
+  function defaultAlert(opt: AlertOptions): void {
+    createAlertHandler(opt);
+  }
```

Comment:
these method should specify `type` at least. When user calls `warning`, he should not pass type